common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/NativeAuthOAuth2Configuration.kt

```diff
@@ -47,22 +48,176 @@ class NativeAuthOAuth2Configuration(
 ) : MicrosoftStsOAuth2Configuration() {
 
     private val TAG = NativeAuthOAuth2Configuration::class.java.simpleName
+
     companion object {
         //Base url for the mock API to make Native Auth calls. See the swagger at
         // https://native-ux-mock-api.azurewebsites.net/doc#/ for all possible urls
         private const val MOCK_API_URL_WITH_NATIVE_AUTH_TENANT = "https://native-ux-mock-api.azurewebsites.net/lumonconvergedps.onmicrosoft.com"
-    }
 
+        private const val SIGNUP_START_ENDPOINT_SUFFIX = "/signup/v1.0/start"
+        private const val SIGNUP_CHALLENGE_ENDPOINT_SUFFIX = "/signup/v1.0/challenge"
+        private const val SIGNUP_CONTINUE_ENDPOINT_SUFFIX = "/signup/v1.0/continue"
+        private const val RESET_PASSWORD_START_ENDPOINT_SUFFIX = "/resetpassword/v1.0/start"
+        private const val RESET_PASSWORD_CHALLENGE_ENDPOINT_SUFFIX = "/resetpassword/v1.0/challenge"
+        private const val RESET_PASSWORD_CONTINUE_ENDPOINT_SUFFIX = "/resetpassword/v1.0/continue"
+        private const val RESET_PASSWORD_SUBMIT_ENDPOINT_SUFFIX = "/resetpassword/v1.0/submit"
+        private const val RESET_PASSWORD_COMPLETE_ENDPOINT_SUFFIX = "/resetpassword/v1.0/poll_completion"
+        private const val SIGN_IN_INITIATE_ENDPOINT_SUFFIX = "/oauth2/v2.0/initiate"
+        private const val SIGN_IN_CHALLENGE_ENDPOINT_SUFFIX = "/oauth2/v2.0/challenge"
+        private const val SIGN_IN_TOKEN_ENDPOINT_SUFFIX = "/oauth2/v2.0/token"
+    }
 
     override fun getAuthorityUrl(): URL {
         return if (useMockApiForNativeAuth) {
-            // TODO return real authorityUrl once we move away from using mock APIs
             URL(MOCK_API_URL_WITH_NATIVE_AUTH_TENANT)
         } else {
             authorityUrl
         }
     }
 
+    /**
+     * Get the endpoint to be used for making a signup/start request.
+     *
+     * @return URL the endpoint
+     */
+    fun getSignUpStartEndpoint(): URL {
```

Comment:
For the sake of making this PR a bit smaller, could we hold off on adding the SignUp and ResetPassword endpoint methods?

Reply:
I meant for this particular file. Thanks for confirming the other files don't include signup resetpassword methods